[2025-05-21T09:40:12.932+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data scheduled__2025-05-20T09:38:43.708532+00:00 [queued]>
[2025-05-21T09:40:12.943+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data scheduled__2025-05-20T09:38:43.708532+00:00 [queued]>
[2025-05-21T09:40:12.944+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2025-05-21T09:40:12.960+0000] {taskinstance.py:1380} INFO - Executing <Task(BashOperator): transform_taxi_data> on 2025-05-20 09:38:43.708532+00:00
[2025-05-21T09:40:12.967+0000] {standard_task_runner.py:57} INFO - Started process 663 to run task
[2025-05-21T09:40:12.970+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'yellow_taxi_batch_pipeline', 'transform_taxi_data', 'scheduled__2025-05-20T09:38:43.708532+00:00', '--job-id', '76', '--raw', '--subdir', 'DAGS_FOLDER/taxi_batch_dag.py', '--cfg-path', '/tmp/tmpd3df619h']
[2025-05-21T09:40:12.973+0000] {standard_task_runner.py:85} INFO - Job 76: Subtask transform_taxi_data
[2025-05-21T09:40:13.000+0000] {logging_mixin.py:151} WARNING - /home/airflow/.local/lib/python3.8/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-05-21T09:40:13.057+0000] {task_command.py:415} INFO - Running <TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data scheduled__2025-05-20T09:38:43.708532+00:00 [running]> on host 00732dad519c
[2025-05-21T09:40:13.165+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='yellow_taxi_batch_pipeline' AIRFLOW_CTX_TASK_ID='transform_taxi_data' AIRFLOW_CTX_EXECUTION_DATE='2025-05-20T09:38:43.708532+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-05-20T09:38:43.708532+00:00'
[2025-05-21T09:40:13.168+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-05-21T09:40:13.169+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\n    spark-submit         --master spark://spark-master:7077         --jars /opt/airflow/dags/jars/***ql-42.5.0.jar         --packages org.apache.hadoop:hadoop-aws:3.3.1         /opt/airflow/dags/scripts/taxi_transform.py\n    ']
[2025-05-21T09:40:13.180+0000] {subprocess.py:86} INFO - Output:
[2025-05-21T09:40:13.190+0000] {subprocess.py:93} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-05-21T09:40:15.230+0000] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.0.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-05-21T09:40:15.385+0000] {subprocess.py:93} INFO - Ivy Default Cache set to: /home/airflow/.ivy2/cache
[2025-05-21T09:40:15.386+0000] {subprocess.py:93} INFO - The jars for the packages stored in: /home/airflow/.ivy2/jars
[2025-05-21T09:40:15.390+0000] {subprocess.py:93} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2025-05-21T09:40:15.391+0000] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-b82842d7-f6b2-4935-97b2-dce8fd3cae19;1.0
[2025-05-21T09:40:15.392+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-05-21T09:40:17.187+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-aws;3.3.1 in central
[2025-05-21T09:40:17.790+0000] {subprocess.py:93} INFO - 	found com.amazonaws#aws-java-sdk-bundle;1.11.901 in central
[2025-05-21T09:40:18.996+0000] {subprocess.py:93} INFO - 	found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2025-05-21T09:40:19.041+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/3.3.1/hadoop-aws-3.3.1.jar ...
[2025-05-21T09:40:19.128+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.hadoop#hadoop-aws;3.3.1!hadoop-aws.jar (102ms)
[2025-05-21T09:40:19.144+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.11.901/aws-java-sdk-bundle-1.11.901.jar ...
[2025-05-21T09:40:34.292+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] com.amazonaws#aws-java-sdk-bundle;1.11.901!aws-java-sdk-bundle.jar (15162ms)
[2025-05-21T09:40:34.309+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/wildfly/openssl/wildfly-openssl/1.0.7.Final/wildfly-openssl-1.0.7.Final.jar ...
[2025-05-21T09:40:34.374+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.wildfly.openssl#wildfly-openssl;1.0.7.Final!wildfly-openssl.jar (80ms)
[2025-05-21T09:40:34.374+0000] {subprocess.py:93} INFO - :: resolution report :: resolve 3633ms :: artifacts dl 15350ms
[2025-05-21T09:40:34.375+0000] {subprocess.py:93} INFO - 	:: modules in use:
[2025-05-21T09:40:34.376+0000] {subprocess.py:93} INFO - 	com.amazonaws#aws-java-sdk-bundle;1.11.901 from central in [default]
[2025-05-21T09:40:34.376+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-aws;3.3.1 from central in [default]
[2025-05-21T09:40:34.376+0000] {subprocess.py:93} INFO - 	org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2025-05-21T09:40:34.377+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-21T09:40:34.377+0000] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2025-05-21T09:40:34.378+0000] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-05-21T09:40:34.378+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-21T09:40:34.378+0000] {subprocess.py:93} INFO - 	|      default     |   3   |   3   |   3   |   0   ||   3   |   3   |
[2025-05-21T09:40:34.379+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-21T09:40:34.380+0000] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-b82842d7-f6b2-4935-97b2-dce8fd3cae19
[2025-05-21T09:40:34.380+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-05-21T09:40:34.586+0000] {subprocess.py:93} INFO - 	3 artifacts copied, 0 already retrieved (189078kB/205ms)
[2025-05-21T09:40:34.908+0000] {subprocess.py:93} INFO - 25/05/21 09:40:34 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-05-21T09:40:35.142+0000] {subprocess.py:93} INFO - 25/05/21 09:40:35 WARN DependencyUtils: Local jar /opt/airflow/dags/jars/***ql-42.5.0.jar does not exist, skipping.
[2025-05-21T09:40:36.939+0000] {subprocess.py:93} INFO - 25/05/21 09:40:36 INFO SparkContext: Running Spark version 3.3.1
[2025-05-21T09:40:36.978+0000] {subprocess.py:93} INFO - 25/05/21 09:40:36 INFO ResourceUtils: ==============================================================
[2025-05-21T09:40:36.979+0000] {subprocess.py:93} INFO - 25/05/21 09:40:36 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-05-21T09:40:36.980+0000] {subprocess.py:93} INFO - 25/05/21 09:40:36 INFO ResourceUtils: ==============================================================
[2025-05-21T09:40:36.981+0000] {subprocess.py:93} INFO - 25/05/21 09:40:36 INFO SparkContext: Submitted application: YellowTaxiTransform
[2025-05-21T09:40:37.013+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-05-21T09:40:37.025+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO ResourceProfile: Limiting resource is cpu
[2025-05-21T09:40:37.026+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-05-21T09:40:37.109+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SecurityManager: Changing view acls to: airflow
[2025-05-21T09:40:37.111+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SecurityManager: Changing modify acls to: airflow
[2025-05-21T09:40:37.112+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SecurityManager: Changing view acls groups to:
[2025-05-21T09:40:37.113+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SecurityManager: Changing modify acls groups to:
[2025-05-21T09:40:37.114+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(airflow); groups with view permissions: Set(); users  with modify permissions: Set(airflow); groups with modify permissions: Set()
[2025-05-21T09:40:37.473+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO Utils: Successfully started service 'sparkDriver' on port 37869.
[2025-05-21T09:40:37.516+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SparkEnv: Registering MapOutputTracker
[2025-05-21T09:40:37.567+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SparkEnv: Registering BlockManagerMaster
[2025-05-21T09:40:37.587+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-05-21T09:40:37.588+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-05-21T09:40:37.593+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-05-21T09:40:37.624+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-ff39f21f-1265-4d99-a060-dc65392f6594
[2025-05-21T09:40:37.657+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-05-21T09:40:37.696+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-05-21T09:40:37.915+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-05-21T09:40:37.955+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 ERROR SparkContext: Failed to add /opt/spark/jars/***ql-42.5.0.jar to Spark environment
[2025-05-21T09:40:37.956+0000] {subprocess.py:93} INFO - java.io.FileNotFoundException: Jar /opt/spark/jars/***ql-42.5.0.jar not found
[2025-05-21T09:40:37.956+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:1949)
[2025-05-21T09:40:37.957+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.addJar(SparkContext.scala:2004)
[2025-05-21T09:40:37.958+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$new$12(SparkContext.scala:507)
[2025-05-21T09:40:37.958+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$new$12$adapted(SparkContext.scala:507)
[2025-05-21T09:40:37.959+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-05-21T09:40:37.959+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-05-21T09:40:37.960+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-05-21T09:40:37.960+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.<init>(SparkContext.scala:507)
[2025-05-21T09:40:37.961+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-05-21T09:40:37.961+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-05-21T09:40:37.962+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2025-05-21T09:40:37.962+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-05-21T09:40:37.962+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[2025-05-21T09:40:37.963+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-05-21T09:40:37.964+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-05-21T09:40:37.964+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:238)
[2025-05-21T09:40:37.965+0000] {subprocess.py:93} INFO - 	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-05-21T09:40:37.965+0000] {subprocess.py:93} INFO - 	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-05-21T09:40:37.966+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-05-21T09:40:37.966+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-05-21T09:40:37.966+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-21T09:40:37.967+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/org.apache.hadoop_hadoop-aws-3.3.1.jar at spark://00732dad519c:37869/files/org.apache.hadoop_hadoop-aws-3.3.1.jar with timestamp 1747820436917
[2025-05-21T09:40:37.967+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO Utils: Copying /home/airflow/.ivy2/jars/org.apache.hadoop_hadoop-aws-3.3.1.jar to /tmp/spark-463f9fe0-988e-410f-81b4-b6e0614166c9/userFiles-e5c57c50-3165-405a-8b1d-fb6fda6b3637/org.apache.hadoop_hadoop-aws-3.3.1.jar
[2025-05-21T09:40:37.974+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar at spark://00732dad519c:37869/files/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar with timestamp 1747820436917
[2025-05-21T09:40:37.975+0000] {subprocess.py:93} INFO - 25/05/21 09:40:37 INFO Utils: Copying /home/airflow/.ivy2/jars/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar to /tmp/spark-463f9fe0-988e-410f-81b4-b6e0614166c9/userFiles-e5c57c50-3165-405a-8b1d-fb6fda6b3637/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar
[2025-05-21T09:40:38.234+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar at spark://00732dad519c:37869/files/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar with timestamp 1747820436917
[2025-05-21T09:40:38.235+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO Utils: Copying /home/airflow/.ivy2/jars/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar to /tmp/spark-463f9fe0-988e-410f-81b4-b6e0614166c9/userFiles-e5c57c50-3165-405a-8b1d-fb6fda6b3637/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar
[2025-05-21T09:40:38.398+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-05-21T09:40:38.462+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.4:7077 after 34 ms (0 ms spent in bootstraps)
[2025-05-21T09:40:38.601+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250521094038-0002
[2025-05-21T09:40:38.609+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250521094038-0002/0 on worker-20250521093802-172.18.0.7-38743 (172.18.0.7:38743) with 2 core(s)
[2025-05-21T09:40:38.620+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO StandaloneSchedulerBackend: Granted executor ID app-20250521094038-0002/0 on hostPort 172.18.0.7:38743 with 2 core(s), 1024.0 MiB RAM
[2025-05-21T09:40:38.621+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 40887.
[2025-05-21T09:40:38.622+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO NettyBlockTransferService: Server created on 00732dad519c:40887
[2025-05-21T09:40:38.623+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-05-21T09:40:38.640+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 00732dad519c, 40887, None)
[2025-05-21T09:40:38.646+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO BlockManagerMasterEndpoint: Registering block manager 00732dad519c:40887 with 434.4 MiB RAM, BlockManagerId(driver, 00732dad519c, 40887, None)
[2025-05-21T09:40:38.657+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 00732dad519c, 40887, None)
[2025-05-21T09:40:38.664+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 00732dad519c, 40887, None)
[2025-05-21T09:40:38.674+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250521094038-0002/0 is now RUNNING
[2025-05-21T09:40:38.950+0000] {subprocess.py:93} INFO - 25/05/21 09:40:38 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-05-21T09:40:39.260+0000] {subprocess.py:93} INFO - 25/05/21 09:40:39 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-05-21T09:40:39.263+0000] {subprocess.py:93} INFO - 25/05/21 09:40:39 INFO SharedState: Warehouse path is 'file:/tmp/airflowtmp5xw__fri/spark-warehouse'.
[2025-05-21T09:40:40.572+0000] {subprocess.py:93} INFO - 2025-05-21 09:40:40,571 - INFO - Loading Yellow Taxi data from MinIO
[2025-05-21T09:40:40.883+0000] {subprocess.py:93} INFO - 25/05/21 09:40:40 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-05-21T09:40:40.915+0000] {subprocess.py:93} INFO - 25/05/21 09:40:40 INFO MetricsSystemImpl: Scheduled Metric snapshot period at 10 second(s).
[2025-05-21T09:40:40.916+0000] {subprocess.py:93} INFO - 25/05/21 09:40:40 INFO MetricsSystemImpl: s3a-file-system metrics system started
[2025-05-21T09:40:42.837+0000] {subprocess.py:93} INFO - 25/05/21 09:40:42 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.18.0.7:54984) with ID 0,  ResourceProfileId 0
[2025-05-21T09:40:42.923+0000] {subprocess.py:93} INFO - 25/05/21 09:40:42 INFO BlockManagerMasterEndpoint: Registering block manager 172.18.0.7:36037 with 366.3 MiB RAM, BlockManagerId(0, 172.18.0.7, 36037, None)
[2025-05-21T09:40:43.998+0000] {subprocess.py:93} INFO - 25/05/21 09:40:43 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: s3a://nyc-taxi-data/yellow_taxi/yellow_tripdata_2022-*.parquet.
[2025-05-21T09:40:44.007+0000] {subprocess.py:93} INFO - java.nio.file.AccessDeniedException: s3a://nyc-taxi-data/yellow_taxi/yellow_tripdata_2022-*.parquet: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:40:44.012+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:206)
[2025-05-21T09:40:44.017+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:170)
[2025-05-21T09:40:44.018+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3351)
[2025-05-21T09:40:44.020+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.innerGetFileStatus(S3AFileSystem.java:3185)
[2025-05-21T09:40:44.021+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.isDirectory(S3AFileSystem.java:4277)
[2025-05-21T09:40:44.025+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:54)
[2025-05-21T09:40:44.029+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:370)
[2025-05-21T09:40:44.032+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:228)
[2025-05-21T09:40:44.035+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:210)
[2025-05-21T09:40:44.037+0000] {subprocess.py:93} INFO - 	at scala.Option.getOrElse(Option.scala:189)
[2025-05-21T09:40:44.039+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:210)
[2025-05-21T09:40:44.041+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.parquet(DataFrameReader.scala:562)
[2025-05-21T09:40:44.042+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-05-21T09:40:44.045+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2025-05-21T09:40:44.046+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-05-21T09:40:44.048+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2025-05-21T09:40:44.049+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-05-21T09:40:44.051+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-05-21T09:40:44.053+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-05-21T09:40:44.057+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-05-21T09:40:44.063+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-05-21T09:40:44.065+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-05-21T09:40:44.066+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-05-21T09:40:44.069+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-21T09:40:44.071+0000] {subprocess.py:93} INFO - Caused by: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:40:44.072+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:216)
[2025-05-21T09:40:44.072+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1257)
[2025-05-21T09:40:44.073+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:833)
[2025-05-21T09:40:44.074+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:783)
[2025-05-21T09:40:44.075+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
[2025-05-21T09:40:44.076+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
[2025-05-21T09:40:44.077+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
[2025-05-21T09:40:44.078+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
[2025-05-21T09:40:44.079+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
[2025-05-21T09:40:44.081+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
[2025-05-21T09:40:44.081+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5227)
[2025-05-21T09:40:44.088+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:6189)
[2025-05-21T09:40:44.089+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:6162)
[2025-05-21T09:40:44.090+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5211)
[2025-05-21T09:40:44.094+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5173)
[2025-05-21T09:40:44.097+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5167)
[2025-05-21T09:40:44.097+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.listObjectsV2(AmazonS3Client.java:963)
[2025-05-21T09:40:44.099+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$listObjects$7(S3AFileSystem.java:2116)
[2025-05-21T09:40:44.099+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
[2025-05-21T09:40:44.102+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:412)
[2025-05-21T09:40:44.103+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:375)
[2025-05-21T09:40:44.104+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listObjects(S3AFileSystem.java:2107)
[2025-05-21T09:40:44.104+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3322)
[2025-05-21T09:40:44.105+0000] {subprocess.py:93} INFO - 	... 21 more
[2025-05-21T09:40:44.106+0000] {subprocess.py:93} INFO - Caused by: com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:40:44.106+0000] {subprocess.py:93} INFO - 	at com.amazonaws.auth.EnvironmentVariableCredentialsProvider.getCredentials(EnvironmentVariableCredentialsProvider.java:50)
[2025-05-21T09:40:44.110+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:177)
[2025-05-21T09:40:44.112+0000] {subprocess.py:93} INFO - 	... 43 more
[2025-05-21T09:40:46.278+0000] {subprocess.py:93} INFO - 2025-05-21 09:40:46,277 - ERROR - ETL process failed: An error occurred while calling o31.parquet.
[2025-05-21T09:40:46.279+0000] {subprocess.py:93} INFO - : java.nio.file.AccessDeniedException: s3a://nyc-taxi-data/yellow_taxi: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:40:46.279+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:206)
[2025-05-21T09:40:46.280+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.once(Invoker.java:117)
[2025-05-21T09:40:46.280+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listStatus(S3AFileSystem.java:2810)
[2025-05-21T09:40:46.281+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.listStatus(Globber.java:128)
[2025-05-21T09:40:46.282+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.doGlob(Globber.java:291)
[2025-05-21T09:40:46.283+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.glob(Globber.java:202)
[2025-05-21T09:40:46.283+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.globStatus(S3AFileSystem.java:4253)
[2025-05-21T09:40:46.284+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.globStatus(S3AFileSystem.java:4233)
[2025-05-21T09:40:46.285+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkHadoopUtil.globPath(SparkHadoopUtil.scala:253)
[2025-05-21T09:40:46.286+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$3(DataSource.scala:765)
[2025-05-21T09:40:46.286+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$.$anonfun$parmap$2(ThreadUtils.scala:372)
[2025-05-21T09:40:46.287+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[2025-05-21T09:40:46.288+0000] {subprocess.py:93} INFO - 	at scala.util.Success.$anonfun$map$1(Try.scala:255)
[2025-05-21T09:40:46.289+0000] {subprocess.py:93} INFO - 	at scala.util.Success.map(Try.scala:213)
[2025-05-21T09:40:46.291+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[2025-05-21T09:40:46.291+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[2025-05-21T09:40:46.292+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-05-21T09:40:46.294+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-05-21T09:40:46.295+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
[2025-05-21T09:40:46.296+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[2025-05-21T09:40:46.296+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
[2025-05-21T09:40:46.297+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
[2025-05-21T09:40:46.297+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
[2025-05-21T09:40:46.298+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
[2025-05-21T09:40:46.298+0000] {subprocess.py:93} INFO - Caused by: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:40:46.299+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:216)
[2025-05-21T09:40:46.299+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1257)
[2025-05-21T09:40:46.300+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:833)
[2025-05-21T09:40:46.300+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:783)
[2025-05-21T09:40:46.300+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
[2025-05-21T09:40:46.301+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
[2025-05-21T09:40:46.301+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
[2025-05-21T09:40:46.302+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
[2025-05-21T09:40:46.302+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
[2025-05-21T09:40:46.307+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
[2025-05-21T09:40:46.307+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5227)
[2025-05-21T09:40:46.308+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:6189)
[2025-05-21T09:40:46.309+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:6162)
[2025-05-21T09:40:46.309+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5211)
[2025-05-21T09:40:46.310+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5173)
[2025-05-21T09:40:46.310+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5167)
[2025-05-21T09:40:46.311+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.listObjectsV2(AmazonS3Client.java:963)
[2025-05-21T09:40:46.312+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$listObjects$7(S3AFileSystem.java:2116)
[2025-05-21T09:40:46.312+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
[2025-05-21T09:40:46.313+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:412)
[2025-05-21T09:40:46.313+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:375)
[2025-05-21T09:40:46.314+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listObjects(S3AFileSystem.java:2107)
[2025-05-21T09:40:46.314+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem$ListingOperationCallbacksImpl.lambda$listObjectsAsync$0(S3AFileSystem.java:1750)
[2025-05-21T09:40:46.315+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.impl.CallableSupplier.get(CallableSupplier.java:62)
[2025-05-21T09:40:46.315+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
[2025-05-21T09:40:46.315+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2025-05-21T09:40:46.316+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2025-05-21T09:40:46.316+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-21T09:40:46.317+0000] {subprocess.py:93} INFO - Caused by: com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:40:46.318+0000] {subprocess.py:93} INFO - 	at com.amazonaws.auth.EnvironmentVariableCredentialsProvider.getCredentials(EnvironmentVariableCredentialsProvider.java:50)
[2025-05-21T09:40:46.318+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:177)
[2025-05-21T09:40:46.319+0000] {subprocess.py:93} INFO - 	... 27 more
[2025-05-21T09:40:46.319+0000] {subprocess.py:93} INFO - 
[2025-05-21T09:40:46.320+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO SparkUI: Stopped Spark web UI at http://00732dad519c:4040
[2025-05-21T09:40:46.321+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-05-21T09:40:46.323+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2025-05-21T09:40:46.343+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-05-21T09:40:46.386+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO MemoryStore: MemoryStore cleared
[2025-05-21T09:40:46.387+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO BlockManager: BlockManager stopped
[2025-05-21T09:40:46.400+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-05-21T09:40:46.404+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-05-21T09:40:46.465+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO SparkContext: Successfully stopped SparkContext
[2025-05-21T09:40:46.617+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO ShutdownHookManager: Shutdown hook called
[2025-05-21T09:40:46.618+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-463f9fe0-988e-410f-81b4-b6e0614166c9
[2025-05-21T09:40:46.622+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-463f9fe0-988e-410f-81b4-b6e0614166c9/pyspark-ea803b2e-ac84-4470-8f16-ae62821bb20f
[2025-05-21T09:40:46.628+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-a9206833-3a1c-488f-a81c-81950cd08a6a
[2025-05-21T09:40:46.637+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2025-05-21T09:40:46.638+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO MetricsSystemImpl: s3a-file-system metrics system stopped.
[2025-05-21T09:40:46.639+0000] {subprocess.py:93} INFO - 25/05/21 09:40:46 INFO MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2025-05-21T09:40:46.695+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2025-05-21T09:40:46.734+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=yellow_taxi_batch_pipeline, task_id=transform_taxi_data, execution_date=20250520T093843, start_date=20250521T094012, end_date=20250521T094046
[2025-05-21T09:40:46.778+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-05-21T09:40:46.805+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
